samples/javascript_nodejs/09.message-routing/dialogs/greeting/greeting.js

```diff
@@ -125,61 +122,60 @@ class Greeting extends ComponentDialog {
      *
      * Having all the data we need, simply display a summary back to the user.
      *
-     * @param {DialogContext} dc context for this dialog
      * @param {WaterfallStepContext} step contextual information for the current step being executed
      */
-    async displayGreetingStateStep(dc, step) {
+    async displayGreetingStateStep(step) {
         // Save city, if prompted for
-        const greetingState = await this.greetingStateAccessor.get(dc.context);
+        const greetingState = await this.greetingStateAccessor.get(step.context);
         if (greetingState.city === undefined && step.result) {
             let lowerCaseCity = step.result;
             // capitalize and set city
             greetingState.city = lowerCaseCity.charAt(0).toUpperCase() + lowerCaseCity.substr(1);
-            await this.greetingStateAccessor.set(dc.context, greetingState);
-            await this.userState.saveChanges(dc.context);
+            await this.greetingStateAccessor.set(step.context, greetingState);
+            await this.userState.saveChanges(step.context);
         }
-        return await this.greetUser(dc);
+        return await this.greetUser(step);
     }
     /**
      * Validator function to verify that user name meets required constraints.
-     * 
-     * @param {DialogContext} context for this dialog
+     *
      * @param {PromptValidatorContext} prompt context for this prompt
      */
-    async validateName(context, prompt) {
+    async validateName(prompt) {
         // Validate that the user entered a minimum lenght for their name
+        prompt.activeDialog = await this.greetingStateAccessor;
+        prompt.context = await prompt.context;
         const value = (prompt.recognized.value || '').trim();
         if (value.length >= NAME_LENGTH_MIN) {
-            prompt.end(value);
+            return true;
```

Comment:
Please add a comment here specifying why you're returning `true`. While the code in the `else` block is correct, please set an explicit `return false;` and explain why that needs to occur as well.